backend/src/api.py

```diff
@@ -330,34 +331,36 @@ async def store_random_review(feedback_input: Request) -> OkResponse:
     return OkResponse()
 
 
-def _make_tpm_tar_file_from_results() -> Iterator[Path]:
-    if not FEEDBACK_DIR:
-        raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail="No data found")
-
-    with get_temporary_dir() as tmp_dir:
-        tar_path = make_tar(
-            f"results-{int(datetime.now().timestamp())}.tar.gz", Path(FEEDBACK_DIR), tmp_dir
-        )
-        try:
-            yield tar_path
-        finally:
-            os.unlink(tar_path)
+@app.get("/download")
+def download_results():
+    """
+    Download all results we have as a tar.gz archive.
 
+    We create a temporary directory and store the archive in there.
+    After the client browser gets the whole file, we delete our temp
+    file and directory using the cleanup() method as a backgroun task.
 
-@app.get("/download", response_class=StreamingResponse)
-def download_results(_tar_path=Depends(_make_tpm_tar_file_from_results)):
-    def iter_large_file(file_name: Path):
-        with open(file_name, mode="rb") as file:
-            yield from file
+    This function was rewritten from an async implementataion that stopped working
+    (probably after an update to fastapi and starlette).
```

Comment:
correct, the [dependency injection](https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-with-yield/) I wondered why it stopped working... It just go back to the yielding function at the beginning of processing the endpoint (right after return SomeFastAPIThingy), previously it was at the end - so after StreamingResponse did its thing. I thought this behaviour was feature... it seems it was a bug :Dd

Reply:
I don't know honestly :D and didn't have the patience to debug the async implementation. Once I found the FileResponse class I realized I can remove a lot of the code :)